Test/Framework/Visual/GraphicsDeviceTest.cs

```diff
@@ -111,6 +111,66 @@ public void DrawIndexedPrimitivesParameterValidation()
         }
 
         [Test]
+        public void DrawIndexedPrimitivesParameterValidation2()
+        {
+            Game.DrawWith += (sender, e) =>
+            {
+                var vertexBuffer = new VertexBuffer(
+                    Game.GraphicsDevice, VertexPositionColorTexture.VertexDeclaration,
+                    3, BufferUsage.None);
+                var indexBuffer = new IndexBuffer(
+                    Game.GraphicsDevice, IndexElementSize.SixteenBits, 
+                    3, BufferUsage.None);
+
+                // No vertex shader or pixel shader.
+                Assert.Throws<InvalidOperationException>(() => Game.GraphicsDevice.DrawIndexedPrimitives(PrimitiveType.TriangleList, 0, 0, 1));
+
+                new BasicEffect(Game.GraphicsDevice).CurrentTechnique.Passes[0].Apply();
+
+                // No vertexBuffer.
+                Assert.Throws<InvalidOperationException>(() => Game.GraphicsDevice.DrawIndexedPrimitives(PrimitiveType.TriangleList, 0, 0, 1));
+
+                Game.GraphicsDevice.SetVertexBuffer(vertexBuffer);
+
+                // No indexBuffer.
+                Assert.Throws<InvalidOperationException>(() => Game.GraphicsDevice.DrawIndexedPrimitives(PrimitiveType.TriangleList, 0, 0, 1));
+
+                Game.GraphicsDevice.Indices = indexBuffer;
+
+                // Success - "normal" usage.
+                Assert.DoesNotThrow(() => Game.GraphicsDevice.DrawIndexedPrimitives(PrimitiveType.TriangleList, 0, 0, 1));
+
+                // XNA doesn't do upfront parameter validation on the Assert.DoesNotThrow tests,
+                // but it *sometimes* fails later with an AccessViolationException, so we can't actually
+                // run these tests as part of the XNA test suite.
+
+                // baseVertex too small / large.
+#if !XNA
+                Assert.DoesNotThrow(() => Game.GraphicsDevice.DrawIndexedPrimitives(PrimitiveType.TriangleList, -1, 0, 1));
+                Assert.DoesNotThrow(() => Game.GraphicsDevice.DrawIndexedPrimitives(PrimitiveType.TriangleList, 3, 0, 1));
+#endif
```

Comment:
Looks like unused check. However this test will fail if PlatformDrawIndexedPrimitives throws exception. Thats the only reason to keep it up I guess.